src/git_credential_msal/main.py

```diff
@@ -134,31 +134,56 @@ def extract_entra_ids_from_git_config(helper_pairs: dict[str, str]) -> tuple[str
     return (msal_client_id, msal_tenant_id)
 
 
-def get_msal_cache(name: str) -> SerializableTokenCache:
+def get_msal_cache_insecure(name: str, allow_insecure: bool) -> str:
+    if not allow_insecure:
+        return None
+
+    msal_cache_name = f"msal_cache_{name}"
+    msal_cache_path = os.path.join(cache_dir, msal_cache_name)
+    try:
+        with open(msal_cache_path, "r") as f:
+            return f.read()
+    except:
+        return None
+
+
+def get_msal_cache(name: str, allow_insecure: bool) -> SerializableTokenCache:
     cache = SerializableTokenCache()
     data = None
     try:
         data = keyring.get_password("system", name)
     except keyring.errors.NoKeyringError:
-        pass
+        data = get_msal_cache_insecure(name, allow_insecure)
```

Comment:
Would it make sense to move this out of the `except` block and just do
```
if not data and allow_insecure:
    data = get_msal_cache_insecure(name)
```
That way, if someone stores the msal cache insecurely and then later logs into a desktop environment that has Keyring but it doesn't have a stored msal token, it can still find the insecure one?

Reply:
I was thinking about this too and wasn't sure what would be the better design choice. Will support this model. However, if someone installs a keyring, they may not invoke the `--insecure` flag. Would it make sense to still read from an already written insecure cache entry in that situation? Basically omit the `and allow_insecure`.

Reply:
I think it would make sense to allow reading an insecure credential even without `--insecure`. Worse-case it just isn't valid, right? I guess it might be weird to have it read an insecure credential and then later write it to Keyring, leaving stale credentials in the cache. But as long as stale credentials are silently invalidated if they don't work it's probably fine?

Reply:
Yeah, that's exactly the line of thinking that went through my mind as well.